sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobServiceClientBuilder.java

```diff
@@ -374,6 +408,21 @@ public BlobServiceClientBuilder retryOptions(RequestRetryOptions retryOptions) {
         return this;
     }
 
+    /**
+     * Sets the request retry options for all the requests made through the client.
+     *
+     * Consider using {@link #retryOptions(RequestRetryOptions)} to also set storage specific options.
+     *
+     * @param retryOptions {@link RetryOptions}.
+     * @return the updated BlobServiceClientBuilder object
+     * @throws NullPointerException If {@code retryOptions} is {@code null}.
+     */
+    @Override
+    public BlobServiceClientBuilder retryOptions(RetryOptions retryOptions) {
+        Objects.requireNonNull(retryOptions, "'retryOptions' cannot be null.");
+        return this.retryOptions(RequestRetryOptions.fromRetryOptions(retryOptions, null, null));
```

Comment:
We should do a similar validation for Storage too to ensure only one of RetryOptions and RequestRetryOptions is set as we do in other builders for policy and options.